pkg/sync/syncWorker.go

```diff
@@ -389,9 +398,15 @@ func (s *syncWorker) listenToStream(
 				zap.String("peer", node.HttpAddress),
 				zap.Any("numEnvelopes", len(envs.Envelopes)),
 			)
+
 			for _, env := range envs.Envelopes {
-				// ignore errors and proceed with the next message
-				_ = s.validateAndInsertEnvelope(originatorStream, env)
+				// Retry errors that may be transient up to 5X.
+				// If the error is permanent, we swallow it in the attemptEnvelopeProcessing function and allow the stream
+				// to continue. If it is transient and fails 5X, we return the error and halt the stream.
```

Comment:
it won't actually halt the stream. The top level retry will just retry again and keep failing infinitely

Reply:
Maybe halt is too strong a word. Disconnect would be better.

Returning an error here will disconnect from the other node and then it will reconnect (with a backoff) infinitely